provider/resource_keycloak_saml_client.go

```diff
@@ -93,6 +94,12 @@ func resourceKeycloakSamlClient() *schema.Resource {
 				Optional:     true,
 				ValidateFunc: validation.StringInSlice(keycloakSamlClientSignatureAlgorithms, false),
 			},
+			"signature_key_name": {
```

Comment:
since we never call a `data.Get` or `data.Set` for this schema attribute, it is essentially ignored.  take a look at the `mapToSamlClientFromData` and `mapToDataFromSamlClient` functions to see where these should be added.